### ISSUE TITLE
header objects being ignored

### ISSUE THREAD

**Issue by Allc9001** [Rhosys/openapi-explorer#135]
I'm trying to use openapi-explorer using openapi spec version 3.0.1. One of my headers is an object (rather than primimative) and it looks like this header is being ignored. Below is a snippet of my spec which shows this issue. The login endpoint needs 2 headers in the request - realm and x-api-key. While the realm header is populated as expected in the curl example the other header (x-api-key) appears to be ignored. According to the spec a header can be an object:

https://spec.openapis.org/oas/v3.0.1#parameter-object

Finally, if I take my spec snippet and plug it into https://editor.swagger.io/ it works as expected.

What do you think ? Should this work or am I doing something wrong ?

Here is the snippet of my openapi spec to show the issue:

```
{
  "openapi" : "3.0.1",
  "info" : {
      "title": "Title",
      "version": "Version"
  },
  "paths" : {
    "/api/{ver}/login" : {
      "post" : {
        "tags" : [ "User" ],
        "summary" : "Login with secret",
        "description" : "Allow a user to login with a valid id and secret.",
        "operationId" : "login_1",
        "parameters" : [ {
          "name" : "ver",
          "in" : "path",
          "description" : "The version of the endpoint",
          "required" : true,
          "schema" : {
            "type" : "string"
          },
          "example" : "v2"
        }, {
          "name" : "Realm",
          "in" : "header",
          "required" : true,
          "schema" : {
            "type" : "string"
          }
        }, {
          "name" : "x-api-key",
          "in" : "header",
          "schema" : {
            "$ref" : "#/components/schemas/ApiKey"
          },
          "explode": true
        } ],
        "requestBody" : {
          "content" : {
            "application/json" : {
              "schema" : {
                "$ref" : "#/components/schemas/LoginRequest"
              }
            }
          },
          "required" : true
        },
        "responses" : {
          "201" : {
            "description" : "A user was successfully authenticated."
          },
          "404" : {
            "description" : "A user failed to login."
          }
        }
      }
    }
  },
  "components" : {
    "schemas" : {
      "ApiKey" : {
        "type" : "object",
        "properties" : {
          "key" : {
            "type" : "string"
          }
        }
      },
      "LoginRequest" : {
        "type" : "object",
        "properties" : {
          "id" : {
            "type" : "string"
          },
          "idPresent" : {
            "type" : "boolean"
          },
          "passwordPresent" : {
            "type" : "boolean"
          },
          "secret" : {
            "type" : "string"
          }
        }
      }
    }
  }
}
```

**Comment by wparad** [comment]
It's totally possible objects in headers aren't being handled perfectly. It is just not showing up in the `curl` display, or is it also not available in the UI (schema / input)? Is it actually still sent to the registered endpoint? Or are all four places broken?

**Comment by Allc9001** [comment]
I've only observed it not being compiled into the curl command. I will give you a better answer later today when I do some more testing.

**Comment by wparad** [comment]
So I actually took a quick look, I thought it was showing in the curl at least as far as I could see, but wasn't being handled correctly in some edge cases. In any case we are in the process of fixing that,  but confirmation on your side of what the expected output would be will help us ensure that we put in the right fix.

**Comment by wparad** [comment]
Resolved! The new version is `1.0.498`.

![image](https://user-images.githubusercontent.com/5056218/220433913-b7fed94d-72dd-4382-8563-16903eed0bdc.png)

![image](https://user-images.githubusercontent.com/5056218/220434120-6d08a11b-91f4-4e8a-bb09-6b28961c70b6.png)
